### PR TITLE
[Admin]Login

### PR DESCRIPTION
**Pull request by khang-sun**
## Related Tickets
- [#49404](https://edu-redmine.sun-asterisk.vn/issues/49404)
## WHAT (optional)
- Authenticate for admin page
## HOW
- Add function to check if an user is admin or not
## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

**Comment by khang-sun** [comment]
ready

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -25,4 +25,18 @@ def logged_in_user
...
+    flash[:danger] = t ".admin_warning"
+    redirect_to root_path
+  end
+
+  def not_admin
+    return unless is_admin?
+
+    flash[:danger] = t ".warning"
+    redirect_to admin_root_path
+  end
```
Comment (anlvq-1058) [inline app/controllers/application_controller.rb:41]: Gộp lại làm 1 hàm thôi em

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -19,6 +19,12 @@ def logged_in?
     current_user.present?
   end
 
+  def is_admin?
+    return unless logged_in?
+
+    current_user.role == "admin"
```
Comment (anlvq-1058) [inline app/helpers/sessions_helper.rb:25]: current_user.admin?

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by khang-sun** [comment]
ready

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**